File: ExpressionTreeTransform.Tests/CompilerGenerated.cs
using System;
using System.Linq.Expressions;
using Xunit;
using static ExpressionTreeTransform.Util.Globals;

namespace ExpressionTreeTransform.Tests {
    public class CompilerGenerated {
        private void buildAssert<T>(Expression<Func<T>>expr, string csharp, string vb) {
            var testCSharpCode = expr.ToCode(CSharp);
            var testVBCode = expr.ToCode(VisualBasic);
            Assert.True(testCSharpCode == csharp);
            Assert.True(testVBCode == vb);
        }

        [Fact]
        public void ReturnBooleanTruel() => buildAssert(() => true, "() => true", "Function() True");

        [Fact]
        public void ReturnBooleanFalse() => buildAssert(() => false, "() => false", "Function() False");

        [Fact]
        public void ReturnMemberAccess() => buildAssert(() => "abcd".Length, "() => \"abcd\".Length", "Function() \"abcd\".Length");

        [Fact]
        public void ReturnObjectCreation() => buildAssert(() => new DateTime(1980, 1, 1), "() => new DateTime(1980, 1, 1)", "Function() New Date(1980, 1, 1)");
    }
}

File: ExpressionTreeTransform.Tests.VB/CompilerGenerated.vb
Imports System.Linq.Expressions
Imports Xunit
Imports ExpressionTreeTransform.Util.Globals

Namespace ExpressionTreeTransform.Tests
    Public Class CompilerGenerated
        Private Sub BuildAssert(Of T)(expr As Expression(Of Func(Of T)), CSharpResult As String, VBResult As String)
            Dim testCSharpCode = expr.ToCode(CSharp)
            Dim testVBCode = expr.ToCode(VisualBasic)
            Assert.True(testCSharpCode = CSharpResult)
            Assert.True(testVBCode = VBResult)
        End Sub

        <Fact>
        Sub ReturnBooleanTrue()
            BuildAssert(Function() True, "() => true", "Function() True")
        End Sub

        <Fact>
        Sub ReturnBooleanFalse()
            BuildAssert(Function() False, "() => false", "Function() False")
        End Sub

        <Fact>
        Sub ReturnObjectCreation()
            BuildAssert(Function() New Date(1980, 1, 1), "() => new DateTime(1981, 1, 1)", "Function() New Date(1980, 1, 1)")
        End Sub

        <Fact>
        Sub ReturnDateLiteral()
            BuildAssert(Function() #1/1/1980#, "() => #DateTime", "Function() #1/1/1980")
        End Sub
    End Class
End Namespace

